source/acquaman/BioXAS/BioXASMainXASScanActionController.cpp

```diff
@@ -127,7 +149,39 @@ AMAction3* BioXASMainXASScanActionController::createCleanupActions()
 
 void BioXASMainXASScanActionController::buildScanControllerImplementation()
 {
+	// Create data sources for the dark current corrected measurements of each able detector in the configuration.
+	// Add data sources to list of sources associated with this scan.
+
+	int dwellTimeIndex = scan_->indexOfDataSource(BioXASMainBeamline::bioXAS()->dwellTimeDetector()->name());
+
+	if (dwellTimeIndex != -1) {
+		AMDataSource *dwellTimeSource = scan_->dataSourceAt(dwellTimeIndex);
+
+		for (int i = 0, size = configuration_->detectorConfigurations().count(); i < size; i++) {
+			AMDetector *detector = AMBeamline::bl()->exposedDetectorByInfo(configuration_->detectorConfigurations().at(i));
+
+			if (detector) {
```

Comment:
This is an interesting state check.  This implies that a detector that was added to the configuration was not available in the beamline.  Maybe this should have an else with an error monitor to notify if this ever happens?